tests/test_sqlite.py

```diff
@@ -210,17 +210,45 @@ def makeRegistry(self) -> Registry:
 class SqliteFileRegistryNameKeyCollMgrTestCase(SqliteFileRegistryTests, unittest.TestCase):
     """Tests for `Registry` backed by a SQLite file-based database.
 
-    This test case uses NameKeyCollectionManager.
+    This test case uses NameKeyCollectionManager and
+    ByDimensionsDatasetRecordStorageManager.
     """
     collectionsManager = "lsst.daf.butler.registry.collections.nameKey.NameKeyCollectionManager"
+    datasetsManager = \
+        "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManager"
 
 
 class SqliteFileRegistrySynthIntKeyCollMgrTestCase(SqliteFileRegistryTests, unittest.TestCase):
     """Tests for `Registry` backed by a SQLite file-based database.
 
-    This test case uses SynthIntKeyCollectionManager.
+    This test case uses SynthIntKeyCollectionManager and
+    ByDimensionsDatasetRecordStorageManager.
     """
     collectionsManager = "lsst.daf.butler.registry.collections.synthIntKey.SynthIntKeyCollectionManager"
+    datasetsManager = \
+        "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManager"
+
+
+class SqliteFileRegistryNameKeyCollMgrUUIDTestCase(SqliteFileRegistryTests, unittest.TestCase):
+    """Tests for `Registry` backed by a SQLite file-based database.
+
+    This test case uses NameKeyCollectionManager and
+    ByDimensionsDatasetRecordStorageManagerUUID.
+    """
+    collectionsManager = "lsst.daf.butler.registry.collections.nameKey.NameKeyCollectionManager"
+    datasetsManager = \
+        "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManagerUUID"
+
+
+class SqliteFileRegistrySynthIntKeyCollMgrUUIDTestCase(SqliteFileRegistryTests, unittest.TestCase):
+    """Tests for `Registry` backed by a SQLite file-based database.
+
+    This test case uses SynthIntKeyCollectionManager and
+    ByDimensionsDatasetRecordStorageManagerUUID.
+    """
+    collectionsManager = "lsst.daf.butler.registry.collections.synthIntKey.SynthIntKeyCollectionManager"
+    datasetsManager = \
+        "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManagerUUID"
```

Comment:
I don't know if we want to start down this path of combinatorial manager testing, at least in terms of running all of these by default.

Probably best to punt this to another ticket (and get @timj thinking about it), but I think we should start marking the non-default-manager tests with something like
```
@unittest.skipUnless("DAF_BUTLER_TEST_EXHAUSTIVE" in os.environ, "Skipping optional tests")
```
and the figure out where we want that environment variable to be set.

Reply:
I imagine we always want to fully test the current managers but for the deprecated managers we only want to test basic functionality. Eventually we hope to disable old managers when DMCCB allows us to stop supporting old deployments.

Reply:
The alternative may be to test each manager separately and isolation and only test full registry with a default one. But I'm not sure how difficult would be to test manager code in isolation as it depends on bunch of things from registry.